Added additional features to preserve atom order and template labels when generating reactions

alongd opened the pull request:
### Motivation or Problem
To assist in atom-mapping RMG reactions, some relatively minor modifications were made.

### Description of Changes
-  Use the save_order argument in generate_resonance_structures() 
- Added labeled_reactant/product_atom attribute to TemplateReaction
- Added relabel_atoms and delete_labels arguments to methods of KineticsFamily. These are used to optionally avoid relabeling atoms and to optionally avoid deleting atom labels, respectively. The original behavior is preserved.

In all cases, the original behavior was preserved.

### Testing
A test of retaining atom labels in a template reaction was added.

mjohnson541 commented on rmgpy/data/kinetics/family.py:
```diff
@@ -1754,11 +1757,13 @@ def _create_reaction(self, reactants, products, is_forward):
             for label, atom in reactant.get_all_labeled_atoms().items():
-                labeled_atoms.append((label, atom))
-        reaction.labeledAtoms = labeled_atoms
+                reaction.labeled_reactant_atoms.append((label, atom))
```
Would this not be easier to work with as a dictionary than a list of tuples?

alongd replied:
Absolutely agree. I'll make the change.

mjohnson541 reviewed (commented):
This looks mostly good, my only question is about whether the labeled_reactant_atoms and labeled_product_atoms should be dictionaries instead of lists of tuples.

alongd commented:
@mjohnson541, I modified the attribute to be a dict, now called `.labled_atoms` with `reactants` and `products` keys, respective values are `{label: atom}` dicts.
I removed a commit where the species constraints debug message was deleted (so this debug message is back now) because it gave some trouble in the tests.
I took the liberty to squash my fixup changes since they weren't major, I hope that's OK.
Thanks for the feedback!
